src/hssm/hssm.py

```diff
@@ -592,35 +614,120 @@ def sample_posterior_predictive(
+        ):
+            # Reassign posterior to sub-sampled version
+            setattr(idata_copy, "posterior", idata["posterior"].isel(draw=draws))
+
+        if kind == "pps":
+            # If we run kind == 'pps' we actually run the observation RV
+            if safe_mode:
+                # safe mode splits the draws into chunks of 10 to avoid
+                # memory issues (TODO: Figure out the source of memory issues)
+                split_draws = _split_array(
+                    idata_copy["posterior"].draw.values, divisor=10
+                )
+
+                posterior_predictive_list = []
+                for samples_tmp in split_draws:
+                    tmp_posterior = idata["posterior"].sel(draw=samples_tmp)
+                    setattr(idata_copy, "posterior", tmp_posterior)
+                    self.model.predict(
+                        idata_copy, kind, data, True, include_group_specific
+                    )
+                    posterior_predictive_list.append(idata_copy["posterior_predictive"])
+
+                if inplace:
+                    idata.add_groups(
+                        posterior_predictive=xr.concat(
+                            posterior_predictive_list, dim="draw"
+                        )
+                    )
+                    # for inplace, we don't return anything
+                    return None
+                else:
+                    # Reassign original posterior to idata_copy
+                    setattr(idata_copy, "posterior", idata["posterior"])
+                    # Add new posterior predictive group to idata_copy
+                    del idata_copy["posterior_predictive"]
+                    idata_copy.add_groups(
+                        posterior_predictive=xr.concat(
+                            posterior_predictive_list, dim="draw"
+                        )
+                    )
+                    return idata_copy
+            elif inplace:
+                # If not safe-mode
+                # We call .predict() directly without any
+                # chunking of data.
+
+                # .predict() is called on the copy of idata
+                # since we still subsampled (or assigned) the draws
                 self.model.predict(idata_copy, kind, data, True, include_group_specific)
+
+                # posterior predictive group added to idata
                 idata.add_groups(
                     posterior_predictive=idata_copy["posterior_predictive"]
                 )
-
+                # don't return anything if inplace
                 return None
-
+            else:
+                # Not safe mode and not inplace
+                # Function acts as very thin wrapper around
+                # .predict(). It just operates on the
+                # idata_copy object
+                return self.model.predict(
+                    idata_copy, kind, data, inplace, include_group_specific
+                )
```

Comment:
This if block looks slightly confusing. I think I understand what you mean, but would 

```python
if safe_mode:
    if inplace:
        ...
    else:
        ...
else:
    if inplace:
         ...
    else:
         ...
```
be more readable?

Reply:
changed